graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,969 @@
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+import matplotlib as mpl
+from matplotlib.colors import ListedColormap
+
+
+def _check_data(data):
+    if not isinstance(data, np.ndarray):
+        raise TypeError("Data must be a np.ndarray.")
+    if data.ndim != 2:
+        raise ValueError("Data must have dimension 2.")
+
+
+def _check_item_in_meta(meta, item, name):
+    if item is None:
+        return []
+    if isinstance(item, str):
+        item = [item]
+    else:
+        try:
+            iter(item)
+        except TypeError:
+            msg = (
+                f"{name} must be an iterable or string corresponding to columns in meta"
+            )
+            raise TypeError(msg)
+    for col_name in item:
+        if col_name not in meta.columns:
+            if (name == "class_order") and (col_name == "size"):
+                pass
+            else:
+                raise ValueError(f"{col_name} is not a column in the meta dataframe.")
+
+    return item
+
+
+def _check_length(item, name, length):
+    if length != len(item):
+        raise ValueError(
+            f"Length of {name} must be the same as corresponding data axis"
+        )
+
+
+def _check_plotting_kws(palette, color_class):
+    if isinstance(palette, str) or isinstance(palette, dict):
+        palette = [palette]
+    elif len(palette) != len(color_class) and len(palette) != 1:
+        raise ValueError("There is not enough palette value for each color class")
+
+    return palette
+
+
+def _check_sorting_kws(
+    length, meta, group_class, class_order, item_order, color_class, highlight
+):
+    if isinstance(meta, pd.DataFrame):
+        # if meta is here, than everything else must be column item in meta
+        _check_length(meta, "meta", length)
+        group_class = _check_item_in_meta(meta, group_class, "group_class")
+        class_order = _check_item_in_meta(meta, class_order, "class_order")
+        item_order = _check_item_in_meta(meta, item_order, "item_order")
+        color_class = _check_item_in_meta(meta, color_class, "color_class")
+        highlight = _check_item_in_meta(meta, highlight, "highlight")
+    else:
+        # otherwise, arguments can be a hodgepodge of stuff
+        group_class_meta, group_class = _item_to_df(group_class, "group_class", length)
+        class_order_meta, class_order = _item_to_df(class_order, "class_order", length)
+        item_order_meta, item_order = _item_to_df(item_order, "item_order", length)
+        color_class_meta, color_class = _item_to_df(color_class, "color_class", length)
+        highlight_meta, highlight = _item_to_df(highlight, "highlight", length)
+        metas = []
+        for m in [
+            group_class_meta,
+            class_order_meta,
+            item_order_meta,
+            color_class_meta,
+            highlight_meta,
+        ]:
+            if m is not None:
+                metas.append(m)
+        if len(metas) > 0:
+            meta = pd.concat(metas, axis=1)
```

Comment:
@zeou1 all of this code is to let the user pass arguments for anything that are not references to `meta` but rather just lists or arrays or any iterable. as in, a list of "group_class" labels, for example. does that make sense?

Reply:
Yes, I understanded it, but I didn't know what would be the right data type for these

Reply:
array-like of the same shape as the corresponding matrix axis. 

somewhere in there, I also allow for list of list, which I'm not sure if we'd actually want to support. we should decide and make it clear what we support.

Reply:
this is in response to https://github.com/microsoft/graspologic/pull/546#discussion_r537874963